# Issue 27240: py3: tests/cmdline.py

archive/issues_027003.json:
```json
{
    "assignees": [],
    "body": "<div id=\"comment:0\"></div>\n\nThe file tests/cmdline.py fails a few doctests with Python 3 because it relies on the scripts `sage-rst2sws` and `sage-rst2txt`, which in turn use `sagenb`. Since `sagenb` is not Python 3 compatible, the tests fail.\n\nThe script `sage-rst2txt` is only incompatible in the sense that it raises a deprecation warning, and so we can ignore that. Trying to run `sage-rst2sws` produces a genuine error, from this line in `sagenb.notebook.smtpsend`:\n\n```\nfrom twisted.mail import smtp, relaymanager  # problematic with python 3\n```\nSo ignore the error for `sage-rst2txt` with Python 3, and don't run the tests for `sage-rst2sws` at all with Python 3.\n\nComponent: **python3**\n\nAuthor: **John Palmieri**\n\nBranch/Commit: **[`0dc692e`](https://github.com/sagemath/sagetrac-mirror/commit/0dc692ef8453fc1004218ffae880b96cb066211f)**\n\nReviewer: **Fr\u00e9d\u00e9ric Chapoton**\n\n_Issue created by migration from https://trac.sagemath.org/ticket/27240_\n\n",
    "closed_at": "2019-02-11T23:06:40Z",
    "created_at": "2019-02-09T03:35:04Z",
    "labels": [
        "https://github.com/sagemath/sage/labels/c%3A%20python3",
        "https://github.com/sagemath/sage/labels/p%3A%20minor%20/%204",
        "https://github.com/sagemath/sage/labels/bug"
    ],
    "milestone": "https://github.com/sagemath/sage/milestones/sage-8.7",
    "reactions": [],
    "repository": "https://github.com/sagemath/sage",
    "title": "py3: tests/cmdline.py",
    "type": "issue",
    "updated_at": "2019-02-11T23:06:40Z",
    "url": "https://github.com/sagemath/sage/issues/27240",
    "user": "https://github.com/jhpalmieri"
}
```
<div id="comment:0"></div>

The file tests/cmdline.py fails a few doctests with Python 3 because it relies on the scripts `sage-rst2sws` and `sage-rst2txt`, which in turn use `sagenb`. Since `sagenb` is not Python 3 compatible, the tests fail.

The script `sage-rst2txt` is only incompatible in the sense that it raises a deprecation warning, and so we can ignore that. Trying to run `sage-rst2sws` produces a genuine error, from this line in `sagenb.notebook.smtpsend`:

```
from twisted.mail import smtp, relaymanager  # problematic with python 3
```
So ignore the error for `sage-rst2txt` with Python 3, and don't run the tests for `sage-rst2sws` at all with Python 3.

Component: **python3**

Author: **John Palmieri**

Branch/Commit: **[`0dc692e`](https://github.com/sagemath/sagetrac-mirror/commit/0dc692ef8453fc1004218ffae880b96cb066211f)**

Reviewer: **Frédéric Chapoton**

_Issue created by migration from https://trac.sagemath.org/ticket/27240_





---

archive/issue_events_371607.json:
```json
{
    "actor": "https://github.com/jhpalmieri",
    "created_at": "2019-02-09T03:35:04Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "milestone_number": null,
    "milestone_title": "sage-8.7",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/27240#event-371607"
}
```



---

archive/issue_events_371608.json:
```json
{
    "actor": "https://github.com/jhpalmieri",
    "created_at": "2019-02-09T03:35:04Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "label": "https://github.com/sagemath/sage/labels/c%3A%20python3",
    "label_color": "0000b0",
    "label_name": "c: python3",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/27240#event-371608"
}
```



---

archive/issue_events_371609.json:
```json
{
    "actor": "https://github.com/jhpalmieri",
    "created_at": "2019-02-09T03:35:04Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "label": "https://github.com/sagemath/sage/labels/p%3A%20minor%20/%204",
    "label_color": "ffe799",
    "label_name": "p: minor / 4",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/27240#event-371609"
}
```



---

archive/issue_events_371610.json:
```json
{
    "actor": "https://github.com/jhpalmieri",
    "created_at": "2019-02-09T03:35:04Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "label": "https://github.com/sagemath/sage/labels/bug",
    "label_color": "d73a4a",
    "label_name": "bug",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/27240#event-371610"
}
```



---

archive/issue_comments_420812.json:
```json
{
    "body": "Branch: **[u/jhpalmieri/cmdline-py3](https://github.com/sagemath/sagetrac-mirror/tree/u/jhpalmieri/cmdline-py3)**",
    "created_at": "2019-02-09T03:35:40Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/27240#issuecomment-420812",
    "user": "https://github.com/jhpalmieri"
}
```

Branch: **[u/jhpalmieri/cmdline-py3](https://github.com/sagemath/sagetrac-mirror/tree/u/jhpalmieri/cmdline-py3)**



---

archive/issue_comments_420813.json:
```json
{
    "body": "Commit: **[`d9975ad`](https://github.com/sagemath/sagetrac-mirror/commit/d9975ade72ad57b62763e9466583fd264fc4f1f5)**",
    "created_at": "2019-02-09T03:35:51Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/27240#issuecomment-420813",
    "user": "https://github.com/jhpalmieri"
}
```

Commit: **[`d9975ad`](https://github.com/sagemath/sagetrac-mirror/commit/d9975ade72ad57b62763e9466583fd264fc4f1f5)**



---

archive/issue_events_371611.json:
```json
{
    "actor": "https://github.com/jhpalmieri",
    "created_at": "2019-02-09T03:35:51Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "label": "https://github.com/sagemath/sage/labels/needs%20review",
    "label_color": "7fff00",
    "label_name": "needs review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/27240#event-371611"
}
```



---

archive/issue_comments_420814.json:
```json
{
    "body": "<div id=\"comment:2\"></div>\n\nNew commits:\n<table><tr><td><a href=\"https://github.com/sagemath/sagetrac-mirror/commit/d9975ade72ad57b62763e9466583fd264fc4f1f5\"><code>d9975ad</code></a></td><td><code>trac 27240: py3 doctest fixes for tests/cmdline.py</code></td></tr></table>\n",
    "created_at": "2019-02-09T03:35:51Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/27240#issuecomment-420814",
    "user": "https://github.com/jhpalmieri"
}
```

<div id="comment:2"></div>

New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/d9975ade72ad57b62763e9466583fd264fc4f1f5"><code>d9975ad</code></a></td><td><code>trac 27240: py3 doctest fixes for tests/cmdline.py</code></td></tr></table>




---

archive/issue_comments_420815.json:
```json
{
    "body": "Description changed:\n``````diff\n--- \n+++ \n@@ -5,4 +5,4 @@\n ```\n from twisted.mail import smtp, relaymanager  # problematic with python 3\n ```\n-So ignore the error for `sage-rst2txt` with Python 3, and don't run the test for `sage-rst2sws` at all with Python 3.\n+So ignore the error for `sage-rst2txt` with Python 3, and don't run the tests for `sage-rst2sws` at all with Python 3.\n``````\n",
    "created_at": "2019-02-09T03:36:33Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/27240#issuecomment-420815",
    "user": "https://github.com/jhpalmieri"
}
```

Description changed:
``````diff
--- 
+++ 
@@ -5,4 +5,4 @@
 ```
 from twisted.mail import smtp, relaymanager  # problematic with python 3
 ```
-So ignore the error for `sage-rst2txt` with Python 3, and don't run the test for `sage-rst2sws` at all with Python 3.
+So ignore the error for `sage-rst2txt` with Python 3, and don't run the tests for `sage-rst2sws` at all with Python 3.
``````




---

archive/issue_comments_420816.json:
```json
{
    "body": "<div id=\"comment:4\" align=\"right\">comment:4</div>\n\nlooks good\n\nmaybe we can use the opportunity to fix the pyflakes warnings in the modified file, using\n\n```\nfrom subprocess import Popen, PIPE\n```\n?",
    "created_at": "2019-02-09T19:31:45Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/27240#issuecomment-420816",
    "user": "https://github.com/fchapoton"
}
```

<div id="comment:4" align="right">comment:4</div>

looks good

maybe we can use the opportunity to fix the pyflakes warnings in the modified file, using

```
from subprocess import Popen, PIPE
```
?



---

archive/issue_comments_420817.json:
```json
{
    "body": "Changed commit from **[`d9975ad`](https://github.com/sagemath/sagetrac-mirror/commit/d9975ade72ad57b62763e9466583fd264fc4f1f5)** to **[`0dc692e`](https://github.com/sagemath/sagetrac-mirror/commit/0dc692ef8453fc1004218ffae880b96cb066211f)**",
    "created_at": "2019-02-09T20:02:11Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/27240#issuecomment-420817",
    "user": "https://github.com/sagetrac-git"
}
```

Changed commit from **[`d9975ad`](https://github.com/sagemath/sagetrac-mirror/commit/d9975ade72ad57b62763e9466583fd264fc4f1f5)** to **[`0dc692e`](https://github.com/sagemath/sagetrac-mirror/commit/0dc692ef8453fc1004218ffae880b96cb066211f)**



---

archive/issue_comments_420818.json:
```json
{
    "body": "<div id=\"comment:5\"></div>\n\nBranch pushed to git repo; I updated commit sha1. New commits:\n<table><tr><td><a href=\"https://github.com/sagemath/sagetrac-mirror/commit/0dc692ef8453fc1004218ffae880b96cb066211f\"><code>0dc692e</code></a></td><td><code>trac 27240: fix pyflakes warnings for tests/cmdline.py</code></td></tr></table>\n",
    "created_at": "2019-02-09T20:02:11Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/27240#issuecomment-420818",
    "user": "https://github.com/sagetrac-git"
}
```

<div id="comment:5"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/0dc692ef8453fc1004218ffae880b96cb066211f"><code>0dc692e</code></a></td><td><code>trac 27240: fix pyflakes warnings for tests/cmdline.py</code></td></tr></table>




---

archive/issue_comments_420819.json:
```json
{
    "body": "<div id=\"comment:6\" align=\"right\">comment:6</div>\n\nDone.",
    "created_at": "2019-02-09T20:03:32Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/27240#issuecomment-420819",
    "user": "https://github.com/jhpalmieri"
}
```

<div id="comment:6" align="right">comment:6</div>

Done.



---

archive/issue_comments_420820.json:
```json
{
    "body": "<div id=\"comment:7\" align=\"right\">comment:7</div>\n\nok, thanks",
    "created_at": "2019-02-10T12:45:13Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/27240#issuecomment-420820",
    "user": "https://github.com/fchapoton"
}
```

<div id="comment:7" align="right">comment:7</div>

ok, thanks



---

archive/issue_events_371612.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2019-02-10T12:45:13Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "label": "https://github.com/sagemath/sage/labels/needs%20review",
    "label_color": "7fff00",
    "label_name": "needs review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/27240#event-371612"
}
```



---

archive/issue_events_371613.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2019-02-10T12:45:13Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "label": "https://github.com/sagemath/sage/labels/positive%20review",
    "label_color": "dfffc0",
    "label_name": "positive review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/27240#event-371613"
}
```



---

archive/issue_comments_420821.json:
```json
{
    "body": "Reviewer: **Fr\u00e9d\u00e9ric Chapoton**",
    "created_at": "2019-02-10T12:45:20Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/27240#issuecomment-420821",
    "user": "https://github.com/fchapoton"
}
```

Reviewer: **Frédéric Chapoton**



---

archive/issue_comments_420822.json:
```json
{
    "body": "Changed branch from **[u/jhpalmieri/cmdline-py3](https://github.com/sagemath/sagetrac-mirror/tree/u/jhpalmieri/cmdline-py3)** to **[`0dc692e`](https://github.com/sagemath/sagetrac-mirror/commit/0dc692ef8453fc1004218ffae880b96cb066211f)**",
    "created_at": "2019-02-11T23:06:40Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/27240#issuecomment-420822",
    "user": "https://github.com/vbraun"
}
```

Changed branch from **[u/jhpalmieri/cmdline-py3](https://github.com/sagemath/sagetrac-mirror/tree/u/jhpalmieri/cmdline-py3)** to **[`0dc692e`](https://github.com/sagemath/sagetrac-mirror/commit/0dc692ef8453fc1004218ffae880b96cb066211f)**



---

archive/issue_events_371614.json:
```json
{
    "actor": "https://github.com/vbraun",
    "created_at": "2019-02-11T23:06:40Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "label": "https://github.com/sagemath/sage/labels/positive%20review",
    "label_color": "dfffc0",
    "label_name": "positive review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/27240#event-371614"
}
```



---

archive/issue_events_371615.json:
```json
{
    "actor": "https://github.com/vbraun",
    "commit_id": "13f46acfb6bf558c99f835cc998dcb81104d7539",
    "commit_repository": "https://github.com/sagemath/sage",
    "created_at": "2019-02-11T23:06:40Z",
    "event": "closed",
    "issue": "https://github.com/sagemath/sage/issues/27240",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/27240#event-371615"
}
```
